private/pkg/command/runner.go

```diff
@@ -46,32 +48,26 @@ func newRunner(options ...RunnerOption) *runner {
 }
 
 func (r *runner) Run(ctx context.Context, name string, options ...RunOption) error {
-	runOptions := newRunOptions()
-	for _, option := range options {
-		option(runOptions)
-	}
-	if len(runOptions.env) == 0 {
-		runOptions.env = emptyEnv
-	}
-	if runOptions.stdin == nil {
-		runOptions.stdin = ioextended.DiscardReader
-	}
+	runOptions := newRunOptions(options...)
 	cmd := exec.CommandContext(ctx, name, runOptions.args...)
-	cmd.Env = envSlice(runOptions.env)
-	cmd.Stdin = runOptions.stdin
-	// If Stdout or Stderr are nil, os/exec connects the process output directly
-	// to the null device.
-	cmd.Stdout = runOptions.stdout
-	cmd.Stderr = runOptions.stderr
-	// The default behavior for dir is what we want already, i.e. the current
-	// working directory.
-	cmd.Dir = runOptions.dir
+	runOptions.Apply(cmd)
 	r.semaphoreC <- struct{}{}
 	err := cmd.Run()
 	<-r.semaphoreC
 	return err
 }
 
+func (r *runner) Start(name string, options ...RunOption) (Process, error) {
+	runOptions := newRunOptions(options...)
+	cmd := exec.Command(name, runOptions.args...)
+	runOptions.Apply(cmd)
```

Comment:
This function ignores all parallelism that we set - one of the goals of `Runners` is to limit the number of concurrent tasks that can be sent off. If a caller sets `RunnerWithParallelism`, that should be respected. You'll have to rearchitect this function to accomplish this. You'll have to +1 the semaphore before you start the command, start the command, then kick of a goroutine to handle `cmd.Process.Wait`. The `terminator.Terminate` functionality will need to be rearchitected to handle the results of this `Wait` Once `Wait` returns, the semaphore needs to be -1'ed.